package.json

```diff
@@ -38,6 +38,7 @@
     "glob": "7.1",
     "handlebars": "4.0",
     "helmet": "3.13",
+    "kafka-node": "3.0",
```

Comment:
What are your thoughts on optional provider dependencies? I'd want to make this only a dev dependency since you don't need it unless you use the kafka provider. But, that means that someone wanting to use the kafka provider needs to add it as a dependency.

Reply:
@reblace I wish it were a bit easier to configure dependencies dynamically for situations like this, but in lieu of that, I'm leaning toward leaving them in as a full dependency, since this is a server-side app and we don't need to be as concerned with the package size.

When considering the workflow of optional provider dependencies, I think it might get tricky managing some in devDependencies, some in dependencies, and some in both (considering the forks that we expect).

That said, I'm open to trying out either way and adjusting as needed down the road.

Reply:
Pulling this tread a bit...

This is similar to the discussion around common vs core things. Kafka is not a core dependency of the application because there are ways you can run the app without depending on it. Or, you may want to add your own provider (e.g., activemq, rabbitmq, etc.). The kafka dependency is necessary only to run the tests. Someone who wanted to configure their app to use the kafka provider would add that dependency to "dependencies" so it's part of the production npm install.

To keep things clean (and to keep us honest), I'd suggest we leave the default config in a "reference application" mode with minimal external dependencies (i.e., only MongoDB). We can set up "dependencies" to include only the things needed to run the application in this mode. As part of the README, we can talk about providers and how you need to add any external dependencies that you might introduce with alternate providers to the production dependencies list.

The result of this would be that in cases like this, we would add Kafka as a dependency, but only to the "devDependecies" list because it's only needed to run the tests as part of the build.

I had similar feelings about something like an elasticsearch client. I want to avoid the "out of the box" application from having a lot of explicit external dependencies.

Reply:
Agreed. I'll make the change to move it to "devDependencies" and start adding to our README with those details. Thanks for the suggestion!